### ISSUE TITLE
Setting the repository will be used unfiltered

### ISSUE THREAD

**Issue by sourcerer-mike** [deployphp/deployer#628]
| Q                 | A
| ----------------- | ---
| Issue Type        | Question
| Deployer Version  | 3.2.1
| Local Machine OS  | Ubuntu 15.04
| Remote Machine OS | Ubuntu 15.04

### Description 

GIT says the directory is given but it isn't.
The deploy target is just new and there is no such directory.

### Steps to reproduce

- Complete new deployer project as suggested in your documentation.
- No directory exists
- Owner of the target directory is the correct user.

### Content of `deploy.php`
```php
<?php

// All Deployer recipes are based on `recipe/common.php`.
require 'recipe/symfony.php';

// Define a server for deployment.
// Let's name it "prod" and use port 22.
server( 'prod', 'the.remote', 22 )
	->user( 'ME' )
	->forwardAgent()// You can use identity key, ssh config, or username/password to auth on the server.
	->stage( 'production' )
	->env( 'deploy_path', '/srv/www/test' ); // Define the base path to deploy your project to.

// Specify the repository from which to download your project's code.
// The server needs to have git installed for this to work.
// If you're not using a forward agent, then the server has to be able to clone
// your project from this repository.
set( 'repository', `git config --get -z remote.origin.url` );
```

### Output log

```
$ bin/dep deploy -vvv prod
➤ Executing task deploy:prepare
↳ on [prod]
> echo $0
< bash
< 
> if [ ! -d /srv/www/test ]; then mkdir -p /srv/www/test; fi
> if [ ! -L /srv/www/test/current ] && [ -d /srv/www/test/current ]; then echo true; fi
> cd /srv/www/test && if [ ! -d releases ]; then mkdir releases; fi
> cd /srv/www/test && if [ ! -d shared ]; then mkdir shared; fi
• done on [prod]
✔ Ok [1s 488ms]
➤ Executing task deploy:release
↳ on [prod]
> mkdir /srv/www/test/releases/20160417234518
> cd /srv/www/test && if [ -h release ]; then rm release; fi
> ln -s /srv/www/test/releases/20160417234518 /srv/www/test/release
• done on [prod]
✔ Ok [428ms]
➤ Executing task deploy:update_code
↳ on [prod]
> which git
< /usr/bin/git
< 
> git version
< git version 1.9.1
< 
> find /srv/www/test/releases -maxdepth 1 -mindepth 1 -type d
< /srv/www/test/releases/20160417233429
< /srv/www/test/releases/20160417234518
< /srv/www/test/releases/20160417233800
< /srv/www/test/releases/20160417234157
< /srv/www/test/releases/20160417233604
< /srv/www/test/releases/20160417233311
< 
> readlink /srv/www/test/release
< /srv/www/test/releases/20160417234518
< 
> /usr/bin/git clone  --depth 1 --recursive -q git@bitbucket.org:ME/foo.git /srv/www/test/releases/20160417234518 2>&1

                                                                              
  [RuntimeException]                                                          
  fatal: destination path 'foo' already exists and is not an empty directory.  
                                                                             
```

**Comment by elfet** [comment]
Try run this by hand: `git clone  --depth 1 --recursive -q git@bitbucket.org:ME/foo.git /srv/www/test/releases/20160417234518`

**Comment by sourcerer-mike** [comment]
Oh yeah, I forgot. This works very well when I am on the server (same user).


```
Submodule 'srv/wp-content/plugins/wp-plugin-readme-parser' (https://github.com/markjaquith/WordPress-Plugin-Readme-Parser.git) registered for path 'srv/wp-content/plugins/wp-plugin-readme-parser'
Cloning into 'srv/wp-content/plugins/wp-plugin-readme-parser'...
remote: Counting objects: 70, done.
remote: Total 70 (delta 0), reused 0 (delta 0), pack-reused 70
Unpacking objects: 100% (70/70), done.
Checking connectivity... done.
Submodule path 'srv/wp-content/plugins/wp-plugin-readme-parser': checked out '69ed0bf10d79668a8e0e0f777e1c4033a0339cc4'
```

It's correctly cloned.


```
/srv/www/test/releases/20160418103839$ tree -L 1
.
├── bin
├── composer.json
├── composer.lock
├── srv
└── wp-cli.yml
```

Where can I look up the deploy task and debug it down? 
I don't find the part that sends the git clone to the server.

**Comment by sourcerer-mike** [comment]
Here is the problem, I reword the title later.

I did this to dynamically resolve the git url: 

```
set( 'repository', `git config --get -z remote.origin.url` );
// without zero char will break this too:
// set( 'repository', `git config --get remote.origin.url` );
```

The command send to the server contains either a null-character or a new line like this:


```sh
/usr/bin/git clone  --depth 1 --recursive -q git@bitbucket.org:rmp/cb.git 
 /srv/www/test/releases/20160418105535 2>&1

# or zero char
/usr/bin/git clone  --depth 1 --recursive -q git@bitbucket.org:rmp/cb.git\0 /srv/www/test/releases/20160418105535 2>&1
```

Both make the command end after the remote URL. As a workaround I now do this:


```
set( 'repository', trim( `git config --get remote.origin.url` ) );
```

If you like keep this issue open and move the `trim` (for the repository url) in your libs.

**Comment by oanhnn** [comment]
Thank @sourcerer-mike ,
It will resolved with #629 .
